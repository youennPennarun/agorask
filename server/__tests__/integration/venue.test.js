/* global test expect */
const fetch = require('node-fetch');
const co = require('co');

test('get a venue by Id', () => {
  return co(function* () {
    const query = `
      query q($id: ID!) {
        venue(id: $id)  {
        _id
        name
        foursquareId
        website
        address {
          location
          formatted
        } 
        categories {
          name
          icon
        }
        tasks {
          _id
          title
          nbAnswers
        }
        contact {
          facebookName
          facebook
          twitter
          formattedPhone
          phone
        }
      }
    }
    `;
    const variables = {
      id: '581773476d568ac7723e1bae'
    };
    const response = yield fetch(`http://localhost:3000/graphql`, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({query, variables}),
    });
    return response.json()
  }).then(json => {
    expect(json).toMatchSnapshot();
  });
});
