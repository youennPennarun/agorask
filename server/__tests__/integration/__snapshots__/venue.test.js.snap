exports[`test get a venue by Id 1`] = `
Object {
  "data": Object {
    "venue": Object {
      "_id": "581773476d568ac7723e1bae",
      "address": Object {
        "formatted": Array [
          "17 Ormeau Ave",
          "Belfast",
          "BT2 8HD",
        ],
        "location": Array [
          -5.928990840911865,
          54.592803881688155,
        ],
      },
      "categories": null,
      "contact": Object {
        "facebook": null,
        "facebookName": null,
        "formattedPhone": "028 9032 5968",
        "phone": "02890325968",
        "twitter": null,
      },
      "foursquareId": "4b65d319f964a5203a022be3",
      "name": "The Limelight",
      "tasks": null,
      "website": "http://www.limelightbelfast.com",
    },
  },
  "errors": Array [
    Object {
      "locations": Array [
        Object {
          "column": 9,
          "line": 16,
        },
      ],
      "message": "Cannot read property \'map\' of undefined",
      "path": Array [
        "venue",
        "tasks",
      ],
    },
  ],
}
`;
